firmware/library/L1_Drivers/pwm.hpp

```diff
@@ -11,3 +14,129 @@ class PwmInterface
     virtual void SetFrequency(uint32_t frequency_hz) = 0;
     virtual uint32_t GetFrequency() = 0;
 };
+
+class Pwm : public PwmInterface
+{
+ public:
+    static constexpr uint8_t kPwmFunction = 0b001;
+
+    static constexpr uint8_t kDefaultFunction = 0b000;
+
+    // remove when sysclock is complete
+    static constexpr uint32_t kDefaultClockSpeed = 12'000'000;
+
+    template <unsigned pwm_port, unsigned pwm_pin>
+    static Pwm CreatePwm()
+    {
+        static_assert(1 <= pwm_port && pwm_port <= 2,
+                      "Available PWM ports are only on port 1 or 2.");
+        static_assert(pwm_pin <= 5,
+                      "Available PWM pins only go up to pin 5.");
+        return Pwm(pwm_port, pwm_pin);
+    }
+
+    Pwm(uint8_t vpport, uint8_t vppin):
```

Comment:
Make this a constexpr constructor.

Reply:
Also, add a constructor that accepts a PinConfigure & object for dependency injection.